testsuite/model/pom.xml

```diff
@@ -127,14 +132,28 @@
         <profile>
             <id>jpa-federation+infinispan</id>
```

Comment:
When I run this combination locally I get 
```
Caused by: org.hibernate.HibernateException: The internal connection pool has reached its maximum size and no connection is currently available!
	at org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl$PooledConnections.poll(DriverManagerConnectionProviderImpl.java:322)
	at org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl.getConnection(DriverManagerConnectionProviderImpl.java:189)
	at org.hibernate.internal.NonContextualJdbcConnectionAccess.obtainConnection(NonContextualJdbcConnectionAccess.java:35)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded(LogicalConnectionManagedImpl.java:106)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.getPhysicalConnection(LogicalConnectionManagedImpl.java:136)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.getConnectionForTransactionManagement(LogicalConnectionManagedImpl.java:254)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.begin(LogicalConnectionManagedImpl.java:262)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.begin(JdbcResourceLocalTransactionCoordinatorImpl.java:236)
	at org.hibernate.engine.transaction.internal.TransactionImpl.begin(TransactionImpl.java:86)
	at org.keycloak.connections.jpa.JpaKeycloakTransaction.begin(JpaKeycloakTransaction.java:42)
	at org.keycloak.services.DefaultKeycloakTransactionManager.enlist(DefaultKeycloakTransactionManager.java:54)
	at org.keycloak.connections.jpa.DefaultJpaConnectionProviderFactory.create(DefaultJpaConnectionProviderFactory.java:92)
	at org.keycloak.connections.jpa.DefaultJpaConnectionProviderFactory.create(DefaultJpaConnectionProviderFactory.java:59)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:274)
	at org.keycloak.models.jpa.JpaUserProviderFactory.create(JpaUserProviderFactory.java:51)
	at org.keycloak.models.jpa.JpaUserProviderFactory.create(JpaUserProviderFactory.java:33)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:274)
	at org.keycloak.services.DefaultKeycloakSession.userLocalStorage(DefaultKeycloakSession.java:194)
	at org.keycloak.storage.UserStorageManager.localStorage(UserStorageManager.java:78)
	at org.keycloak.storage.UserStorageManager.lambda$addUser$4(UserStorageManager.java:193)
```

Reply:
Right, this happens sometimes in this settings. I cannot reproduce it reliably and is not caused by this PR. However I agree we should understand the cause. It has never occurred to me without the `infinispan` profile which is making it even more tricky

Reply:
https://issues.redhat.com/browse/KEYCLOAK-16545

Reply:
Thanks @vramik !